frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.tsx

```diff
@@ -104,16 +126,38 @@ export const AddToDashSelectDashModal = ({
         onChange={onDashboardSelected}
         collectionId={collectionId}
         value={mostRecentDashboardQuery.data?.id}
+        onOpenCollectionChange={setOpenCollectionId}
+        shouldFetchDashboards={shouldFetchDashboards}
       />
-      <Link onClick={() => setShouldCreateDashboard(true)} to="">
-        <LinkContent>
-          <Icon name="add" className="mx1" />
-          <h4>{t`Create a new dashboard`}</h4>
-        </LinkContent>
-      </Link>
+      {shouldFetchDashboards && (
+        <Link onClick={() => setShouldCreateDashboard(true)} to="">
+          <LinkContent>
+            <Icon name="add" className="mx1" />
+            <h4>{t`Create a new dashboard`}</h4>
+          </LinkContent>
+        </Link>
+      )}
     </ModalContent>
   );
 };
 
+function useIsCollectionWithinPersonalCollection(
```

Comment:
I opt to duplicate this hook because the common parent for the 2 components that use this hook would be `/frontend/src/metabase/` which is too broad for this specific hook.

I'm worried if I add the hook in `/frontend/src/metabase/hooks` it might not be appropriate because it will only be used in 2 places.